spikeinterface/preprocessing/silence_periods.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+import scipy.interpolate
+
+from spikeinterface.core.core_tools import define_function_from_class
+from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
+
+from ..core import get_random_data_chunks, get_noise_levels
+
+class SilencedPeriodsRecording(BasePreprocessor):
+    """
+    Silence user-defined periods from recording extractor traces. By default, 
+    periods are zeroed-out (mode = 'zeros'). This is only recommended 
+    for traces that are centered around zero (e.g. through a prior highpass
+    filter); if this is not the case, you can also fill the periods with noise.
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor to remove artifacts from
+    list_periods: list of lists/arrays
+        One list per segment of tuples (start_frame, end_frame) to silence
+
+    mode: str
+        Determines what periods are replaced by. Can be one of the following:
+            
+        - 'zeros' (default): Artifacts are replaced by zeros.
+
+        - 'noise': The periods are filled with a gaussion noise that has the
+                   same variance that the one in the recordings, on a per channel
+                   basis
+
+    Returns
+    -------
+    silence_recording: SilencedPeriodsRecording
+        The recording extractor after silencing some periods    
+    """
+    name = 'silence_periods'
+
+    def __init__(self, recording, list_periods, mode='zeros', **random_chunk_kwargs):
```

Comment:
Can you add explicitly the 3 arguments for the random chunk finction? see https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/preprocessing/whiten.py#L18-L23